demo/run_demo.py

```diff
@@ -34,6 +34,13 @@ def supply_chain():
   subprocess.call(write_code_cmd.split())
   copyfile("foo.py", "../functionary_carl/foo.py")
```

Comment:
I think we should work around this by omitting the `z` flag from the `tar` command (just for the sake of the demo). The `z` makes `tar` use `gzip` to compress the archive, which in turn also stores filename and timestamp, in therefore leads to different hashes (see [SO post](http://stackoverflow.com/questions/36464358/why-do-the-md5-hashes-of-two-tarballs-of-the-same-file-differ)).